### ISSUE TITLE
Handling sample generation when using rolling stats: should multiple samples be created?

### ISSUE THREAD

**Issue by stephen-frank** [NREL/Wattile#261]
The following lines of code (from [here](https://github.com/NREL/Wattile/blob/bf9989c34195036595d7510ba1c5e22723fd349c/wattile/buildings_processing.py#L603-L631)) handle the target variable when executing rolling stats:

```
        # adding resampled target back to the dataframe
        target = _resample_data(target, configs)
        data[configs["data_input"]["target_var"]] = target

    ...

    return data

def _resample_data(data, configs):

    # reading configuration parameters.
    bin_interval = configs["data_processing"]["resample"]["bin_interval"]
    bin_closed = configs["data_processing"]["resample"]["bin_closed"]
    bin_label = configs["data_processing"]["resample"]["bin_label"]

    # resample data
    data = data.resample(rule=bin_interval, label=bin_label, closed=bin_closed)

    # take the closest value from the label
    if bin_label == "left":
        data = data.first()
    elif bin_label == "right":
        data = data.last()

    return data
```

If I'm reading this correctly, the top line of code generates exactly one sample of `target` per interval, using either the first or the last value in the interval as governed by `bin_label` and `bin_closed` options. This sample is lined up with the rolling stats for that interval to create data for training. However, if multiple values of `target` are in the interval, the other values are dropped/lost (I think?).

Example: `bin_interval` is 5 minutes, and each 5-min interval has 5 samples of power. The above code would just keep the first (or last) of those 5 samples, correct?

Since we are trying to model quantiles, each of the samples contains useful information and we don't really want to drop them. When using rolling stats, I would think that we can create a larger and more robust data set of samples by combining each value of `target` within the interval with the rolling stats of predictors for the interval. So for this hypothetical input:

![image](https://user-images.githubusercontent.com/1534082/212381672-de392556-adcf-4870-a23d-f394cabe556d.png)

Instead of this output (which is what I think the code does now):

![image](https://user-images.githubusercontent.com/1534082/212381818-32dde13f-47be-4a6a-8c2b-6131e3e10857.png)

You would get this output:

![image](https://user-images.githubusercontent.com/1534082/212381870-e9a95407-b036-4725-a042-65d25ee9b012.png)

...which yields 5x the samples for training and also correctly represents the full variability of the target within the interval.

Interested in thoughts from @buechler314 and @JanghyunJK. This topic can wait for after Janghyun's PTO.

**Comment by stephen-frank** [comment]
I stumbled on this trying to create correct test cases for issue #260. It is not urgent; just something to think about.

**Comment by buechler314** [comment]
I agree in your interpretation of what the code is doing, although, I must admit I've become a little unfamiliar with this section of the repo, so I also run the risk of misinterpreting. I agree, we would be losing useful information in the first approach, and the second approach may be a remedy. It's a little like [bootstrapping](https://en.wikipedia.org/wiki/Bootstrapping_(statistics)). The above comments are from just looking at the data itself.

But if we consider the model architecture, we can ask if this method would actually help the LSTM learn. The latter method implies that we don't have any idea if an input of 2 is any more likely to map to an output of 5 or 7. This might cause the quantile bands to be much larger than is realistic. This is less of a comment about method 1 vs 2 that Steve mentioned, and more about rolling stats in general. I'll keep thinking about this.
